### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by moise-mulungu**
On this milestone, I:
- Created model classes for all entities.
- Set up associations between models.
- Inserted data in the different models
- Created `users,` `posts,` and `comments` in the rails console
- Created custom methods for `Users,` `Posts,` `Comments,` and `Likes` tables
- Updated the README file.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+end
```
Comment (ahmetbozaci) [inline app/models/user.rb]: - [ ] Kindly add associations between user and `likes, comments, posts `models.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+
```
Comment (ahmetbozaci) [inline app/models/post.rb]: - [ ] Kindly add associations between posts and `likes, comments `models.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  def posts_count_updated
+    users.increment!(:posts_counter)
+  end
```
Comment (ahmetbozaci) [inline app/models/post.rb]: ### OPTIONAL

- [ ] You can use [rails active callbacks.](https://guides.rubyonrails.org/active_record_callbacks.html)

``` 
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```
`after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

File: db/migrate/20220707165025_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users, &:timestamps
+  end
+end
```
Comment (ahmetbozaci) [inline db/migrate/20220707165025_create_users.rb]: - [ ] You already create user, comment, post, like model in the last milestone. You don't have to create again. Kindly delete these migration files.

image
image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @moise-mulungu  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- Well done creating the methods according to requirement.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: app/models/user.rb (hunk 1 above)
Comment (DeliceLydia) [reply app/models/user.rb]: Addressed

File: app/models/post.rb (hunk 2 above)
Comment (DeliceLydia) [reply app/models/post.rb]: addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi @moise-mulungu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**